test/importer.spec.js

```diff
@@ -676,5 +676,43 @@ strategies.forEach((strategy) => {
         }
       }
     })
+
+    it('supports passing mtime', async () => {
+      this.timeout(60 * 1000)
+
+      const options = {
+        rawLeaves: true
+      }
+      const now = parseInt(Date.now() / 1000)
+
+      for await (const file of importer([{
+        path: '1.2MiB.txt',
+        content: bigFile,
+        mtime: now
+      }], ipld, options)) {
+        const node = await exporter(file.cid, ipld)
+
+        expect(node.unixfs.mtime).to.equal(now)
+      }
+    })
+
+    it('supports passing mode', async () => {
+      this.timeout(60 * 1000)
+
+      const options = {
+        rawLeaves: true
+      }
+      const mode = parseInt('0111', 8)
+
+      for await (const file of importer([{
+        path: '1.2MiB.txt',
+        content: bigFile,
+        mode
+      }], ipld, options)) {
+        const node = await exporter(file.cid, ipld)
+
+        expect(node.unixfs.mode).to.equal(mode)
+      }
+    })
```

Comment:
Test that a given file gets the default mode and mtime if unspecified?

Test you can add multiple files and they can each have a different mtime/mode and also have the defaults?

Test that you can't add meta to a directory?

Test that the meta isn't inherited by files in deeper directories?

Reply:
Moar tests have been added